data/transactions/logic/debugger.go

```diff
@@ -102,6 +112,16 @@ func makeDebugState(cx *evalContext) DebugState {
 	}
 	cx.debugState.Globals = globals
 
+	// pre-allocate state maps
+	if (cx.runModeFlags & runModeApplication) != 0 {
+		ds.GlobalStateChanges = make(basics.StateDelta)
+
+		locals := 1 + len(cx.Txn.Txn.Accounts) // sender + referenced accounts
```

Comment:
not incorrect since you're only using it for preallocation, but technically there can be duplicate addresses in `cx.Txn.Txn.Accounts`.

Reply:
could you please add a comment about that so it's less confusing for someone reading this code for the first time?

Reply:
added a comment